packages/storybook/src/generators/configuration/util-functions.ts

```diff
@@ -166,15 +166,12 @@ export function configureTsProjectConfig(
 
   if (
     !tsConfigContent?.exclude?.includes('**/*.stories.ts') &&
-    !tsConfigContent?.exclude?.includes('**/*.stories.js')
+    !tsConfigContent?.exclude?.includes('**/*.stories.js') &&
+    !tsConfigContent?.exclude?.includes('**/*.stories.{ts,tsx,js,jsx,mdx}')
   ) {
     tsConfigContent.exclude = [
       ...(tsConfigContent.exclude || []),
-      '**/*.stories.ts',
-      '**/*.stories.js',
-      ...(isFramework('react', schema) || isFramework('react-native', schema)
-        ? ['**/*.stories.jsx', '**/*.stories.tsx']
-        : []),
+      '**/*.stories.{ts,tsx,js,jsx,mdx}',
```

Comment:
Maybe something like 
```typescript
const extensions = isFramework('react', schema) || isFramework('react-native', schema) ? ['ts','tsx','jsx','js'] : ['ts', 'js'];
tsConfigContent.exclude = [
  (...tsConfigContent.exclude || []),      
  `**/*.stories.{${extensions.join(',')}}`,
]
```

Reply:
Yeah, I decided that there's no real value in so much logic behind these. We want to exclude tests, include stories in `.storybook/tsconfig.json` and we want to exclude stories in the project's `tsconfig.json`. No need for framework specific things. That's why I made it all like that.